code/datums/components/hallucinations.dm

```diff
@@ -111,21 +113,26 @@ ABSTRACT_TYPE(/datum/component/hallucination)
 /datum/component/hallucination/random_sound
 	var/list/sound_list
 	var/sound_prob = 10
+	var/min_distance = -1
 
-	Initialize(timeout=30, sound_list=null, sound_prob=10)
+	Initialize(timeout=30, sound_list=null, sound_prob=10, min_distance = -1)
 		.=..()
 		if(. == COMPONENT_INCOMPATIBLE || length(sound_list) == 0)
 			return .
 		src.sound_list = sound_list
 		src.sound_prob = sound_prob
+		src.min_distance = min_distance
 
 
 	do_mob_tick(mob, mult)
 		if(probmult(src.sound_prob))
 			var/atom/origin = parent_mob.loc
 			var/turf/mob_turf = get_turf(parent_mob)
 			if (mob_turf)
-				origin = locate(mob_turf.x + rand(-10,10), mob_turf.y + rand(-10,10), mob_turf.z)
+				if(src.min_distance > 0)
+					origin = locate(mob_turf.x + pick(rand(-10,-src.min_distance),rand(src.min_distance,10)), mob_turf.y + pick(rand(-10,-src.min_distance),rand(src.min_distance,10)), mob_turf.z)
+				else
+					origin = locate(mob_turf.x + rand(-10,10), mob_turf.y + rand(-10,10), mob_turf.z)
```

Comment:
Does this need to be split like this, as opposed to just defaulting `min_distance` to 0, which should match current behaviour?

Reply:
I'll change that since it would indeed match current behavior, I was overthinking it.

Reply:
still need to set the default to `0` instead of `-1`